app/modules/code_provider/github/github_service.py

```
@@ -24,15 +24,40 @@
+async def _get_async_redis_cache():  # noqa: ANN201
+    """Return shared async Redis client for cache; create on first use."""
+    global _async_redis_cache
+    if _async_redis_cache is not None:
+        return _async_redis_cache
+    if redis_async is None or not config_provider.get_redis_url():
+        return None
+    try:
+        _async_redis_cache = redis_async.from_url(
+            config_provider.get_redis_url(), decode_responses=False
+        )
+        return _async_redis_cache
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)